In [26]:
import numpy as np
from math import factorial

In [27]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### Task1 ###

Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

#### Решение ####

In [28]:
arr = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
M = np.sum(arr)/len(arr)
M  

65.3

среднее арифметическое: 65.3

In [29]:
S = (np.sum((arr - M)**2)/(len(arr)-1)) ** (1/2)
S

31.624607341019814

среднее квадратичное отклонение: 31.625

In [30]:
S2 = S ** 2
S2

1000.1157894736843

несмещенная оценка дисперсий: 1000.116

In [31]:
S = (np.sum((arr - M)**2)/(len(arr)))
S

950.11

смещенная оценка дисперсий: 950.11

### Task2 ###

В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

#### Решение ####

1-й случай: из первой корзины выбрали 0 белых мячей, и из 2-й корзины выбрали 3 белых мяча:

In [36]:
p1 = combinations(3, 2)/combinations(8, 2) * combinations(5, 3)/combinations(12, 3)
p1

0.00487012987012987

2-й случай: из первой корзины выбрали 1 белый мяч, и из 2-й корзины выбрали 2 белых мяча:

In [37]:
p2 = combinations(5, 1)/combinations(8, 1) * combinations(5, 2)/combinations(12, 2)
p2

0.0946969696969697

3-й случай: из первой корзины выбрали 2 белых мяча, и из 2-й корзины выбрали 1 белый мяч:

In [38]:
p3 = combinations(5, 2)/combinations(8, 2) * combinations(5, 1)/combinations(12, 1)
p3

0.1488095238095238

In [39]:
p = p1 + p2 + p3
p

0.24837662337662336

Ответ: 25.84%

### Task3 ###

На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом

#### Решение ####

Считаем полную вероятность попадания в мишень:

In [40]:
P_A = (1/3)*0.9 + (1/3) * 0.8 + (1/3)*0.6
P_A

0.7666666666666666

Вероятность того, что выстрелил 1-й спортсмен:

In [41]:
P_B1_A = (1/3) * 0.9/P_A
P_B1_A

0.391304347826087

Ответ а): 39.13%

Вероятность того, что выстрелил 2-й спортсмен:

In [42]:
P_B2_A = (1/3) * 0.8/P_A
P_B2_A

0.3478260869565218

Ответ б): 34.78%

Вероятность того, что выстрелил 3-й спортсмен:

In [43]:
P_B3_A = (1/3) * 0.6/P_A
P_B3_A

0.2608695652173913

Ответ в): 26.09%

### Task4 ###

В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C

#### Решение ####

Считаем полную вероятность сдачи студентом первой сессии:

In [45]:
P_A = (1/4)*0.8 + (1/4)*0.7 + (2/4)*0.9
P_A

0.825

Вероятность того, что сдавший сессию студент с факультета A:

In [46]:
P_B1_A = (1/4)*0.8 / P_A
P_B1_A

0.24242424242424246

Ответ а): 24.24%

Вероятность того, что сдавший сессию студент с факультета B:

In [47]:
P_B2_A = (1/4)*0.7 / P_A
P_B2_A

0.21212121212121213

Ответ б): 21.21%

Вероятность того, что сдавший сессию студент с факультета C:

In [48]:
P_B3_A = (2/4)*0.9 / P_A
P_B3_A

0.5454545454545455

Ответ в):54.55%

### Task5 ###

Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей

#### Решение ####

а)

In [54]:
p3 = 0.1 * 0.2 * 0.25
p3

0.005000000000000001

Ответ а): 0.5%

б) 1-я и 2-я или 2-я и 3-я или 1-я и 3-я

In [55]:
p2 = 0.1 * 0.2 + 0.2 * 0.25 + 0.1 * 0.25
p2

0.095

Ответ б): 9.5%

в) 1-я или 2-я или 3-я

In [56]:
p1 = 0.1 + 0.2 + 0.25
p1

0.55

Ответ в): 55%

г) или вариант б) или вариант в)

In [57]:
p12 = p1 + p2
p12

0.645

Ответ г): 64.5%